docs/reference/upgrade/preparing_to_upgrade.asciidoc

```diff
@@ -0,0 +1,23 @@
+[float]
+=== Preparing to upgrade
+
+Before you start to upgrade your cluster to version {version}.
+
+. Check the <<deprecation-logging, deprecation log>> to see if you are using any
+deprecated features and update your code accordingly.
+
+. Review the <<breaking-changes,breaking changes>> and make any necessary
+changes to your code and configuration for version {version}.
+
+. If you use any plugins, make sure there is a version of each plugin that is
+compatible with {es} version {version}.
+
+. Test the upgrade in an isolated environment before upgrading your production
+cluster.
+
+. <<modules-snapshots,Back up your data by taking a snapshot!>> Once you have
+started to upgrade your cluster to version {version} you must complete the
+upgrade. As soon as the cluster contains nodes of version {version} it may make
+changes to its internal state that cannot be reverted. If you cannot complete
+the upgrade then you must deploy an empty cluster and restore its contents from
```

Comment:
Perhaps you could start the section with this. Currently this feels hidden away in the bullet point about snapshots, whereas this is not only a property of that.

Reply:
Thanks for the suggestion, yes, it looks good there; see http://elasticsearch_47342.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/restart-upgrade.html#restart-upgrade